### DISCUSSION TITLE
Best way to save a custom configuration?

### DISCUSSION THREAD

**Discussion by wespiard** [manna-harbour/miryoku#246] [Q&A] [answered]
I have made some customizations using `custom_rules.mk` and `custom_config.h` and I'm curious: what's the best way to save these? Should I fork `miryoku_qmk`, commit/push the changes, and then pull updates from upstream every once in a while?

I'm pretty familiar with a standard git workflow, but don't have much experience with forks and keeping them updated.

**Comment by manna-harbour** [comment, marked as answer]
Yes, I'd suggest keeping your changes in a new branch in your fork and pulling from the upstream `miryoku` branch periodically. I won't make changes to `custom_config.h` or `custom_rules.mk`,  so you won't have to worry about merge conflicts.

You can easily do this locally in your clone. The process would be the same as the *To update the miryoku branch from Miryoku QMK* step here https://github.com/manna-harbour/miryoku_qmk/tree/miryoku/users/manna-harbour_miryoku#checkout-update-and-merge except you'd make a new branch from the `miryoku` branch first and use that instead. If you also wanted to use the GitHub actions workflow you could push your local branch to your fork and select that branch in the workflow when building.

If you don't want to do anything locally you could make a fork and use the GitHub web interface to pull from upstream. It's easy to keep the local `miryoku` branch in sync with upstream https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork, and since there won't be merge conflicts you could just keep your changes in that branch. It would be better in general to use a new branch for your changes but I suspect pulling into another branch isn't supported directly and would require making and merging pull requests in your fork. Alternatively it's also possible to do the merge only at build time using the build workflow branch merge feature.

**Reply by wespiard** [reply]
Thank you very much!
